### PR TITLE
refactor: remove uint64 to int64 conversions

### LINKED ISSUE #580: Search for all `uint64` to `int64` conversions and remove them
`uint64` to `int64` is dangerous
We may possibly have some because of Unix timestamp conversion
We must remove them

### PR DESCRIPTION
**Pull request by aljo242**
Closes #580

### What does this PR does?

Removes the `uint64` -> `int64` conversions that I could find in our codebase.  Also changes all `BlockHeight` variables to `int64`.  This was the source of some of the conversions and makes sense because the Cosmos SDK itself stores blockheight as int64.

### CODE REVIEW

**Review by Pantani — DISMISSED** [review]
LGMT

File: proto/reward/reward_pool.proto [hunk 1]
```diff
@@ -19,7 +19,7 @@ message RewardPool {
     (gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.Coin",
     (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
   ];
-  uint64 lastRewardHeight = 5;
-  uint64 currentRewardHeight = 6;
+  int64 lastRewardHeight = 5;
+  int64 currentRewardHeight = 6;
```
Comment (lubtd) [inline proto/reward/reward_pool.proto:23]: We have a `Validate` method in `RewardPool`, we should check the values are non-negative

**Review by lubtd — CHANGES_REQUESTED** [review]

File: go.mod [hunk 2]
```diff
@@ -18,8 +18,9 @@ require (
 	github.com/tendermint/starport v0.19.3
 	github.com/tendermint/tendermint v0.34.14
 	github.com/tendermint/tm-db v0.6.4
-	google.golang.org/genproto v0.0.0-20220118154757-00ab72f36ad5
-	google.golang.org/grpc v1.43.0
+	golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd // indirect
```
Comment (giunatale) [inline go.mod:21]:
```suggestion

```

**Review by giunatale — CHANGES_REQUESTED** [review]
agreed with @lubtd on the non-negativity checks

File: x/reward/types/reward_pool.go [hunk 3]
```diff
@@ -3,7 +3,6 @@ package types
 import (
 	"errors"
 	"fmt"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
```
Comment (giunatale) [inline x/reward/types/reward_pool.go:6]: goimports

**Review by giunatale — CHANGES_REQUESTED** [review]

**Review by lubtd — APPROVED** [review]

**Pull request merged**